### PR TITLE
Only draw glyphs on canvas if they are in the font or the font file is missing.

### LINKED ISSUE #6721: à character rendered where it should not be rendered
The PDF file at https://www.dropbox.com/s/ygmic028oykprkw/respond_document_print.pdf?dl=1 has random `à` characters rendered where they should not be.

### PR DESCRIPTION
**Pull request by brendandahl**
Fixes #6721 

I could have sworn there were more PDF's with this issue, but I'm not finding any.

### CODE REVIEW

File: src/core/fonts.js [hunk 1]
```diff
@@ -2836,7 +2840,7 @@ var Font = (function FontClosure() {
           !glyph.matchesForCache(fontChar, unicode, accent, width, vmetric,
                                  operatorListId, isSpace)) {
```
Comment (Snuffleupagus) [inline src/core/fonts.js]: Don't you need to add `isInFont` here as well (and also update the comparison in `Glyph.prototype.matchesForCache`)?

**Comment by Snuffleupagus** [comment]
This also fixes #5642.
It somewhat "regresses" #4402, but given that it has never worked properly it might not matter too much.

**Comment by Snuffleupagus** [comment]
r=me, with the comment addressed and passing test runs.

**Comment by brendandahl** [comment]
Good catch
/botio test

**Comment by brendandahl** [comment]
> It somewhat "regresses" #4402,

I have a fix for this I think, will do in another PR though.

**Pull request merged**